### PR TITLE
Update MPSL

### PR DESCRIPTION
**Pull request by kromancer**
Artifacts extracted from [build #3 of mpsl's release/v1.4-branch](https://jenkins-ncs.nordicsemi.no/job/dragoon/job/dragoon-mpsl/job/release%252Fv1.4-branch/3/)

Corresponding PR in nrf: https://github.com/nrfconnect/sdk-nrf/pull/3492

### CODE REVIEW

File: mpsl/CHANGELOG.rst [hunk 1]
```diff
@@ -9,6 +9,18 @@ Changelog
...
+**********************
+
+Added
+=====
+* Added clock configuration option to not wait for low frequency clock to start (DRGN-14204).
+  See ``mpsl_clock_lfclk_cfg_t::skip_wait_lfclk_started``.
+
+Bugfixes
+========
+* Fixed an issue where the low frequency clock was configured incorrectly when low frequency source was either low  or full swing signal (DRGN-15064).
```
Comment (hurricaneFromMoscow) [inline mpsl/CHANGELOG.rst:22]: Nit-picking. There is an extra whitespace after "low"

File: mpsl/include/mpsl_clock.h [hunk 2]
```diff
@@ -49,38 +50,44 @@ enum MPSL_CLOCK_LF_SRC
...
-                                       is given by @ref MPSL_RECOMMENDED_RC_CTIV
-                                       and @ref MPSL_RECOMMENDED_RC_TEMP_CTIV.
-                                       This sets the calibration interval to 4 seconds and guarantees
-                                       clock calibration every second calibration interval. That is,
-                                       the clock will be calibrated every 8 seconds.
-                                       If the temperature changes more that 0.5 every 4 seconds, the
-                                       clock will be calibrated every 4 seconds. See the
-                                       Product Specification for more information. */
-    uint16_t accuracy_ppm;  /**< Accuracy of the low frequency clock in parts per million (ppm). Default value is @ref MPSL_DEFAULT_CLOCK_ACCURACY_PPM.*/
+  uint8_t source;               /**< LF oscillator clock source, see @ref MPSL_CLOCK_LF_SRC. */
```
Comment (joerchan) [inline mpsl/include/mpsl_clock.h:59]: I find it better to move the documentation above. Then we don't have to keep doing these annoying alignment fixes.
The only change here is aligning to the new variable, right?

**Review by joerchan — APPROVED** [review]

**Comment by kromancer** [comment]
Forced push to fix email compliance check.

File: mpsl/CHANGELOG.rst [hunk 3]
```diff
@@ -9,6 +9,19 @@ Changelog
 
 All notable changes to this project are documented in this file.
 
+nRF Connect SDK v1.4.1
+**********************
+
+Added
+=====
+* Added clock configuration option to not wait for low frequency clock to start (DRGN-14204).
+  See ``mpsl_clock_lfclk_cfg_t::skip_wait_lfclk_started``.
```
Comment (greg-fer) [inline mpsl/CHANGELOG.rst:18]:
```suggestion
* Added a new clock configuration option :c:member:`skip_wait_lfclk_started` in :c:struct:`mpsl_clock_lfclk_cfg_t`, which does not wait for the start of Low Frequency Clock. (DRGN-14204)
```

File: mpsl/CHANGELOG.rst [hunk 4]
```diff
@@ -9,6 +9,19 @@ Changelog
...
+
+Bugfixes
+========
+
+* Fixed an issue where the low frequency clock was configured incorrectly when low frequency source was either low or full swing signal (DRGN-15064).
```
Comment (greg-fer) [inline mpsl/CHANGELOG.rst:23]:
```suggestion
* Fixed an issue where Low Frequency Clock was configured incorrectly when the source configuration signal was set to either External Full swing or External Low swing. (DRGN-15064)
```

**Review by greg-fer — APPROVED** [review]

File: mpsl/include/mpsl_clock.h [hunk 5]
```diff
@@ -49,38 +50,61 @@ enum MPSL_CLOCK_LF_SRC
...
-                                       is given by @ref MPSL_RECOMMENDED_RC_CTIV
-                                       and @ref MPSL_RECOMMENDED_RC_TEMP_CTIV.
-                                       This sets the calibration interval to 4 seconds and guarantees
-                                       clock calibration every second calibration interval. That is,
-                                       the clock will be calibrated every 8 seconds.
-                                       If the temperature changes more that 0.5 every 4 seconds, the
-                                       clock will be calibrated every 4 seconds. See the
-                                       Product Specification for more information. */
-    uint16_t accuracy_ppm;  /**< Accuracy of the low frequency clock in parts per million (ppm). Default value is @ref MPSL_DEFAULT_CLOCK_ACCURACY_PPM.*/
+  /** LF oscillator clock source, see @ref MPSL_CLOCK_LF_SRC. */
```
Comment (joerchan) [inline mpsl/include/mpsl_clock.h:59]: @kromancer Thanks for doing this change, it looks much better now IMO.
I forgot to say though, that now they need an @brief, sorry about that.

**Review by hurricaneFromMoscow — APPROVED** [review]

**Review by rugeGerritsen — APPROVED** [review]

**Pull request merged**